core/lib/types/src/storage_writes_deduplicator.rs

```diff
@@ -113,40 +123,59 @@ impl StorageWritesDeduplicator {
                         panic!("tried removing key: {:?} before insertion", key)
                     });
                     *field_to_change -= 1;
+                    *total_size -= value.size;
+                    if is_write_initial {
+                        *total_size -= BYTES_PER_DERIVED_KEY as usize;
```

Comment:
I don't have any strong opinion about it, but I think it might make code a bit leaner if instead of `metrics.total_size` we stored `metrics.total_updated_values_size` and in order to obtain the full length we could do:

`total_updated_values_size + BYTES_PER_DERIVED_KEY * initial_storage_writes + BYTES_PER_ENUMERATION_INDEX  * repeated_storage_writes`

This way there wouldn't be as many repeated `if`s as now